editing.py

```diff
@@ -175,6 +175,45 @@ def add_artist(self, artist, edit_note, auto=False):
         self.b.submit()
         return self._extract_mbid("artist")
 
+    def add_area(self, area, edit_note, auto=False):
+        # Unlike other methods, this one directly requests page using
+        # mechanize.Requests to overcome mechanize's lack of javascript support
+
+        data_template = {
+            "edit-area.name": area["name"],
+            "edit-area.comment": area["disambiguation"],
+            "edit-area.type_id": area["type"],
+            "edit-area.edit_note": edit_note,
+            "edit-area.iso_3166_1.0": area["iso_3166_1"],
+            "edit-area.iso_3166_2.0": area["iso_3166_2"],
+            "edit-area.iso_3166_3.0": area["iso_3166_3"],
+            "edit-area.url.0.text": area["external_links"]["text"],
+            "edit-area.url.0.link_type_id": area["external_links"]["type"],
```

Comment:
This looks better, but the placeholder `.0.` properties still feel unnecessary. You have to loop over `template.items()` twice (once to remove `None` properties, and another to clobber the "0" properties).

Why not something like
```python
expand_field('iso_3166_1', 'iso_3166_1')
expand_field('external_links', 'url')
```

etc. and just check for `None` plus add the necessary properties in one go? With that in mind, `external_links` should be a list of dicts; right now it looks like you have a dict of lists?